### PR TITLE
Eug

### PR DESCRIPTION
**Pull request by E-Kel**
initial pull request. 
Need advice about the registration test.

### CODE REVIEW

File: src/main/java/constants/Cookie.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package constants;
+
+public class Cookie {
+    public static final String cookie1 = "PrestaShop-a30a9934ef476d11b6cc3c983616e364=gGUlJmz608BKmaascatCZgQJ5Bza%2BaS6Z%2FSlaXDM0sld4fUCdaBmHRIdWKp975yIGByKNxeiYhMLEBY%2BVrNo3U9AFKN02dBKdAcbQaQlvRKGN1O8eYOTZa5%2FHNY9%2F56hphqTxC%2FN9rGaK6MOhfLgLxBgFcAdKLls0NgeTPsnYa%2F4x9upAMBRaknpbdX%2FrZN6quvBxi%2FRFt4uHrxczfxpOGTnRt54QReYLtB%2FfZGUnueTarlk7qBQcBS1%2FCCaKuadwvUDnl6oGroJmhuEZq%2Fi4mEFTgbZRHU8rEbe%2B1sdWAo%3D000219";
```
[the remark this answers is not in the document]
Comment (E-Kel) [reply src/main/java/constants/Cookie.java:4]: Created a special method for cookie fetch.

File: src/main/java/constants/TestDataGeneratorForAddingToCartWithProperties.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    private static Random rnd = new Random();
+    public static String generateIpa() {
+        return String.valueOf(rnd.nextInt(12) + 19);
+    }
+    public static String generateQTY(){
+        return String.valueOf(rnd.nextInt(1000));
+    }
+    private static List<String[]> ls = new ArrayList<>();
+    private static void generate() {
```
[the remark this answers is not in the document]
Comment (E-Kel) [reply src/main/java/constants/TestDataGeneratorForAddingToCartWithProperties.java:17]: fixed

File: src/main/java/constants/TestDataGeneratorForAddingToCartWithProperties.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+    }
+    public static List<String[]> getls(){
+            generate();
+            return ls;
+        }
+
+
+    public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (E-Kel) [reply src/main/java/constants/TestDataGeneratorForAddingToCartWithProperties.java:32]: deleted

File: src/test/java/search/test/DASHSPTESTS.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                        "email_create=test.com&" +
+                        "back=my-account&" +
+                        "token=ce65cefcbafad255f0866d3b32d32058").
+                when().
+                post(BASE_URI + "/index.php?controller=authentication");
+
+        String response1 = response.jsonPath().getString("errors[0]");
+        Boolean response2 = response.jsonPath().get("hasError");
+
+        System.out.println(response.htmlPath().prettyPrint());
```
[the remark this answers is not in the document]
Comment (E-Kel) [reply src/test/java/search/test/DASHSPTESTS.java:31]: it's not my code)

[the remark this answers is not in the document]
Comment (yuliiakholina) [reply src/test/java/search/test/DASHSPTESTS.java:31]: then drop it

File: src/main/java/constants/ResultValues.java [hunk 5]
```diff
@@ -1,24 +0,0 @@
-package constants;
```
Comment (4etverochka) [inline src/main/java/constants/ResultValues.java:1]: What is wrong with this file? Why you deleted it?

File: src/main/java/pages/createAnAccountPage/CreateAnAccountAllFields.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                .formParam("postcode", zip)
+                .formParam("id_country", 21)
+                .formParam("phone_mobile", phone)
+                .formParam("alias", "My+address")
+                .formParam("email_create", "1")
+                .formParam("is_new_customer","1")
+                .formParam("back", "my-account")
+
+                .when()
+                .post(URL.DEFAULT_URL+"?controller=authentication&back=my-account");
```
Comment (4etverochka) [inline src/main/java/pages/createAnAccountPage/CreateAnAccountAllFields.java:47]: Don`t use URL as hardcode

File: src/main/java/pages/signIn/EmailField.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+                        "&" +
+                        "ajax=true" +
+                        "&" +
+                        "email_create=" +testEmail +
+                        "&" +
+                        "back=my-account" +
+                        "&" +
+                        "token=ce65cefcbafad255f0866d3b32d32058")
+                .when()
+                .post(URL.DEFAULT_URL+"?controller=authentication&back=my-account");
```
Comment (4etverochka) [inline src/main/java/pages/signIn/EmailField.java:25]: Same their, hardcode is not the best way.

File: src/test/java/AddToCartWithDiffProperties.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+static void checkThatCartIsEmpty(){
+    Response lastDeletedElement = null;
+    for (int ipa = 19; ipa < 31; ipa++) {
+        lastDeletedElement = deleteProductFromCart(5, ipa);
+    }
+    assertEquals("0", lastDeletedElement.jsonPath().get("nbTotalProducts").toString());
+    allipaSet.clear();
+}
+
+static Stream<Arguments> provideArguments1() {
```
Comment (4etverochka) [inline src/test/java/AddToCartWithDiffProperties.java:49]: Move provide argumets to another class

File: src/test/java/CreateAnAccountPageTest.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        Response response = accountAllFields.createAnAccount(
+                                                                testEmail,
+                                                                pass,
+                                                                fname,
+                                                                lname,
+                                                                address,
+                                                                city,
+                                                                zip,
+                                                                phone);
+     System.out.println(response.getBody().print());
```
Comment (4etverochka) [inline src/test/java/CreateAnAccountPageTest.java:42]: And again 🔢

File: src/test/java/CreateAnAccountPageTest.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+        assertEquals(200, response.statusCode());
+
+
+
+    }
+
+    static Stream<Arguments> provideArgumentsForAccountCreation() {
```
Comment (4etverochka) [inline src/test/java/CreateAnAccountPageTest.java:50]: Same their

File: src/test/java/SignInPageEmailTest.java [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    @MethodSource("provideEmail")
+    void verifyEmailFieldWithValidEmail(String testEmail){
+        EmailField emailField = new EmailField();
+        Response response = emailField.emailValidationOnSignInPage(testEmail);
+        assertEquals(200, response.statusCode());
+        assertEquals(false,response.jsonPath().get("hasError"));
+
+    }
+
+    static Stream<Arguments> provideEmail() {
```
Comment (4etverochka) [inline src/test/java/SignInPageEmailTest.java:28]: And there

**Comment by 4etverochka** [comment]
Pull the last master and merge into our branch!!!

**Pull request closed without merging**